### PR TITLE
Enable Swift 6 language mode for more modules

### PR DESCRIPTION
**Pull request by glbrntt**
Motivation:

v2 will support Swift 6 only. We should enable the Swift 6 language mode
across all v2 modules. This has mostly been done already for non-test 
modules. This change enables it for test modules as well.

Modifications:

- Enable Swift 6 language mode and existential any for test modules

Result:

Closer to fully supporting Swift 6 language mode

### CODE REVIEW

File: Tests/GRPCCodeGenTests/Internal/Renderer/TextBasedRendererTests.swift [hunk 1]
```diff
@@ -41,7 +41,7 @@ final class Test_TextBasedRenderer: XCTestCase {
         Also, bar
         """#
       ),
-      renderedBy: TextBasedRenderer.renderComment,
```
Comment (gjcairo) [inline Tests/GRPCCodeGenTests/Internal/Renderer/TextBasedRendererTests.swift:44]: Curious: what warning were we getting here?

File: Sources/GRPCCore/Streaming/RPCAsyncSequence.swift [hunk 2]
```diff
@@ -16,7 +16,10 @@
 
 /// A type-erasing `AsyncSequence`.
 @available(macOS 15.0, iOS 18.0, watchOS 11.0, tvOS 18.0, visionOS 2.0, *)
-public struct RPCAsyncSequence<Element, Failure: Error>: AsyncSequence, @unchecked Sendable {
+public struct RPCAsyncSequence<
+  Element: Sendable,
+  Failure: Error
+>: AsyncSequence, @unchecked Sendable {
   // @unchecked Sendable is required because 'any' doesn't support composition with primary
   // associated types. (see: https://github.com/swiftlang/swift/issue/63877)
```
[the remark this answers is not in the document]
Comment (glbrntt) [reply Sources/GRPCCore/Streaming/RPCAsyncSequence.swift:24]: Unfortunately not as we then run head first into https://github.com/swiftlang/swift/issues/75242

File: Tests/GRPCCodeGenTests/Internal/Renderer/TextBasedRendererTests.swift (hunk 1 above)
Comment (glbrntt) [reply Tests/GRPCCodeGenTests/Internal/Renderer/TextBasedRendererTests.swift:44]: Implicit capture of self which requires `TextBasedRendered` to be `Sendable` (which it isn't)

File: Sources/GRPCCore/Streaming/RPCAsyncSequence.swift (hunk 2 above)
Comment (glbrntt) [inline Sources/GRPCCore/Streaming/RPCAsyncSequence.swift:24]:
```suggestion
  // associated types. (see: https://github.com/swiftlang/swift/issues/63877)
```

**Review by gjcairo — APPROVED** [review]

**Pull request merged**